httpclient/src/main/java/org/apache/http/impl/conn/BasicHttpClientConnectionManager.java

```diff
@@ -375,7 +375,7 @@ public synchronized void closeIdleConnections(final long idletime, final TimeUni
     @Override
     public synchronized void shutdown() {
         if (this.isShutdown.compareAndSet(false, true)) {
-            shutdownConnection();
```

Comment:
@overquota Should not `#close` do `#closeConnection` and `#shutdown` do `#shutdownConnection`?

Reply:
I think no because `#shutdownConnection` closes connection with `RST` (https://github.com/apache/httpcomponents-core/blob/4.4.x/httpcore/src/main/java/org/apache/http/impl/BHttpConnectionBase.java#L304) - this is not normal tcp connection closing

While `#closeConnection` doing it more correctly (FIN+ACK): https://github.com/apache/httpcomponents-core/blob/4.4.x/httpcore/src/main/java/org/apache/http/impl/BHttpConnectionBase.java#L315

`#shutdown` may do `#shutdownConnection` only if `#shutdownConnection` will do close() on connection instance i think... but if so - i think there are no difference what shutdown method will do close instead of shutdown =)

Reply:
@overquota I am not sure I agree. `#shutdown` is supposed to be used to execute an abnormal connection termination (with `RST`), while `#close` is to be used for orderly connection termination (with `FIN+ACK`).

Reply:
@overquota Actually I just realized that the pooling connection manager does not distinguish between normal or abnormal pool shutdown. Persistent connections get closed out regardless whether the manager gets closed or shut down. For consistency sake I'll make the basic manager behave the same.

Reply:
just for example:
`#shutdown` in [PoolingHttpClientConnectionManager](https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/impl/conn/PoolingHttpClientConnectionManager.java#L410) really do `this.pool.shutdown();`
*BUT*
`this.pool.shutdown` doing [entry.close](https://github.com/apache/httpcomponents-core/blob/4.4.x/httpcore/src/main/java/org/apache/http/pool/AbstractConnPool.java#L148) - which is the same normal `conn.close()`

In addition: [HttpClientBuilder](https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/impl/client/HttpClientBuilder.java#L1232) knows only about `#close`, and on close it is doing... `#shutdown` :) - thats the main reason i've changed `#shutdown` method

In addition 2: [ClientConnectionManager declaration](https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/conn/ClientConnectionManager.java#L112) does not know anything about `#close`, only `#shutdown`. And there are no docs about how connections would be terminated

Do you want me to refactor all these misleadings? No problems, really, but I don't know is it worth it?